oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -213,6 +296,88 @@ int getLifetime() {
         ImmutableMap.of("delegates", this.delegates));
   }
 
+  /**
+   * Returns impersonation account credentials defined by JSON using the format generated by gCloud.
+   * The source credentials in the JSON should be either user account credentials or service account
+   * credentials.
+   *
+   * @param json a map from the JSON representing the credentials
+   * @param transportFactory HTTP transport factory, creates the transport used to get access tokens
+   * @return the credentials defined by the JSON
+   * @throws IOException if the credential cannot be created from the JSON.
+   */
+  static ImpersonatedCredentials fromJson(
+      Map<String, Object> json, HttpTransportFactory transportFactory) throws IOException {
+
+    checkNotNull(json);
+    checkNotNull(transportFactory);
+
+    String serviceAccountImpersonationUrl;
+    List<String> delegates = null;
+    Map<String, Object> sourceCredentialsJson;
+    String sourceCredentialsType;
+    String quotaProjectId;
+    try {
+      serviceAccountImpersonationUrl = (String) json.get("service_account_impersonation_url");
+      if (json.containsKey("delegates")) {
+        delegates = (List<String>) json.get("delegates");
+      }
+      sourceCredentialsJson = (Map<String, Object>) json.get("source_credentials");
+      sourceCredentialsType = (String) sourceCredentialsJson.get("type");
+      quotaProjectId = (String) json.get("quota_project_id");
+    } catch (ClassCastException | NullPointerException e) {
+      throw new CredentialFormatException("An invalid input stream was provided.", e);
+    }
+    String targetPrincipal = extractTargetPrincipal(serviceAccountImpersonationUrl);
```

Comment:
need to handle the IllegalArgumentException  here.

Reply:
Thanks, it is handled.